### DISCUSSION TITLE
installation struggle

### DISCUSSION THREAD

**Discussion by Niki1289** [analogdevicesinc/libiio#1074] [General]
Hello,

I have problems with the installation of libiio. Here is an excerpt of the error message after the make command. I have been stuck for two days.

I have the ADALM-Pluto and I can't use it, can you help me what I'm doing wrong or what dependencies are missing.

Consolidate compiler generated dependencies of target iio-compat
Consolidate compiler generated dependencies of target iiod-responder
[  5%] Built target iio-compat
[  8%] Built target iiod-responder
Consolidate compiler generated dependencies of target iio
[ 50%] Built target iio
Consolidate compiler generated dependencies of target iio_tests_helper
Consolidate compiler generated dependencies of target iiopp-enum
Consolidate compiler generated dependencies of target iiod
[ 55%] Built target iio_tests_helper
[ 57%] Linking CXX executable iiopp-enum
[ 77%] Built target iiod
Consolidate compiler generated dependencies of target iio_info
Consolidate compiler generated dependencies of target iio_stresstest
Consolidate compiler generated dependencies of target iio_genxml
Consolidate compiler generated dependencies of target iio_reg
Consolidate compiler generated dependencies of target iio_attr
Consolidate compiler generated dependencies of target iio_rwdev
[ 81%] Built target iio_reg
[ 91%] Built target iio_stresstest
[ 91%] Built target iio_info
[ 91%] Built target iio_genxml
[ 94%] Built target iio_attr
[ 98%] Built target iio_rwdev
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::create_default_context()':
/home/wizz/libiio/bindings/cpp/./iiopp.h:635: undefined reference to `iio_create_default_context'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::create_scan_context(std::optional<iiopp::cstr>, int)':
/home/wizz/libiio/bindings/cpp/./iiopp.h:748: undefined reference to `iio_create_scan_context'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ScanContext::info_list() const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:729: undefined reference to `iio_scan_context_get_info_list'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ContextInfo::description() const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:694: undefined reference to `iio_context_info_get_description'
/usr/bin/ld:[ 98%] Built target libiio-py
 CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ContextInfo::uri() const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:695: undefined reference to `iio_context_info_get_uri'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::create_scan_block(std::optional<iiopp::cstr>, int)':
/home/wizz/libiio/bindings/cpp/./iiopp.h:786: undefined reference to `iio_create_scan_block'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ScanBlock::ScanBlock(iio_scan_block*)':
/home/wizz/libiio/bindings/cpp/./iiopp.h:769: undefined reference to `iio_scan_block_scan'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ContextInfo::description() const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:694: undefined reference to `iio_context_info_get_description'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ScanBlock::operator[](unsigned int) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:777: undefined reference to `iio_scan_block_get_info'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ContextInfo::uri() const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:695: undefined reference to `iio_context_info_get_uri'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::create_scan_context(std::optional<iiopp::cstr>, int)::{lambda(iiopp::ScanContext*)#1}::operator()(iiopp::ScanContext*) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:755: undefined reference to `iio_scan_context_destroy'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ScanContext::info_list() const::{lambda(iiopp::ScanContext::InfoList*)#1}::operator()(iiopp::ScanContext::InfoList*) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:737: undefined reference to `iio_context_info_list_free'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::create_scan_block(std::optional<iiopp::cstr>, int)::{lambda(iiopp::ScanBlock*)#1}::operator()(iiopp::ScanBlock*) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:793: undefined reference to `iio_scan_block_destroy'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_device, &iio_device_buffer_attr_read, &iio_device_buffer_attr_read_bool, &iio_device_buffer_attr_read_double, &iio_device_buffer_attr_read_longlong, &iio_device_buffer_attr_write, &iio_device_buffer_attr_write_bool, &iio_device_buffer_attr_write_double, &iio_device_buffer_attr_write_longlong>::read(char*, unsigned long) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:228: undefined reference to `iio_device_buffer_attr_read'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_device, &iio_device_debug_attr_read, &iio_device_debug_attr_read_bool, &iio_device_debug_attr_read_double, &iio_device_debug_attr_read_longlong, &iio_device_debug_attr_write, &iio_device_debug_attr_write_bool, &iio_device_debug_attr_write_double, &iio_device_debug_attr_write_longlong>::read(char*, unsigned long) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:228: undefined reference to `iio_device_debug_attr_read'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_device, &iio_device_attr_read, &iio_device_attr_read_bool, &iio_device_attr_read_double, &iio_device_attr_read_longlong, &iio_device_attr_write, &iio_device_attr_write_bool, &iio_device_attr_write_double, &iio_device_attr_write_longlong>::read(char*, unsigned long) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:228: undefined reference to `iio_device_attr_read'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_channel, &iio_channel_attr_read, &iio_channel_attr_read_bool, &iio_channel_attr_read_double, &iio_channel_attr_read_longlong, &iio_channel_attr_write, &iio_channel_attr_write_bool, &iio_channel_attr_write_double, &iio_channel_attr_write_longlong>::read(char*, unsigned long) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:228: undefined reference to `iio_channel_attr_read'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_device, &iio_device_attr_read, &iio_device_attr_read_bool, &iio_device_attr_read_double, &iio_device_attr_read_longlong, &iio_device_attr_write, &iio_device_attr_write_bool, &iio_device_attr_write_double, &iio_device_attr_write_longlong>::write(iiopp::cstr)':
/home/wizz/libiio/bindings/cpp/./iiopp.h:233: undefined reference to `iio_device_attr_write'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_device, &iio_device_debug_attr_read, &iio_device_debug_attr_read_bool, &iio_device_debug_attr_read_double, &iio_device_debug_attr_read_longlong, &iio_device_debug_attr_write, &iio_device_debug_attr_write_bool, &iio_device_debug_attr_write_double, &iio_device_debug_attr_write_longlong>::write(iiopp::cstr)':
/home/wizz/libiio/bindings/cpp/./iiopp.h:233: undefined reference to `iio_device_debug_attr_write'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_device, &iio_device_buffer_attr_read, &iio_device_buffer_attr_read_bool, &iio_device_buffer_attr_read_double, &iio_device_buffer_attr_read_longlong, &iio_device_buffer_attr_write, &iio_device_buffer_attr_write_bool, &iio_device_buffer_attr_write_double, &iio_device_buffer_attr_write_longlong>::write(iiopp::cstr)':
/home/wizz/libiio/bindings/cpp/./iiopp.h:233: undefined reference to `iio_device_buffer_attr_write'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::impl::AttrT<iio_channel, &iio_channel_attr_read, &iio_channel_attr_read_bool, &iio_channel_attr_read_double, &iio_channel_attr_read_longlong, &iio_channel_attr_write, &iio_channel_attr_write_bool, &iio_channel_attr_write_double, &iio_channel_attr_write_longlong>::write(iiopp::cstr)':
/home/wizz/libiio/bindings/cpp/./iiopp.h:233: undefined reference to `iio_channel_attr_write'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::create_scan_block(std::optional<iiopp::cstr>, int)::{lambda(iiopp::ScanBlock*)#1}::operator()(iiopp::ScanBlock*) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:793: undefined reference to `iio_scan_block_destroy'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::ScanContext::info_list() const::{lambda(iiopp::ScanContext::InfoList*)#1}::operator()(iiopp::ScanContext::InfoList*) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:737: undefined reference to `iio_context_info_list_free'
/usr/bin/ld: CMakeFiles/iiopp-enum.dir/examples/iiopp-enum.cpp.o: in function `iiopp::create_scan_context(std::optional<iiopp::cstr>, int)::{lambda(iiopp::ScanContext*)#1}::operator()(iiopp::ScanContext*) const':
/home/wizz/libiio/bindings/cpp/./iiopp.h:755: undefined reference to `iio_scan_context_destroy'
collect2: error: ld returned 1 exit status
make[2]: *** [bindings/cpp/CMakeFiles/iiopp-enum.dir/build.make:98: bindings/cpp/iiopp-enum] Fehler 1
make[1]: *** [CMakeFiles/Makefile2:523: bindings/cpp/CMakeFiles/iiopp-enum.dir/all] Fehler 2
make: *** [Makefile:136: all] Fehler 2

thanks.

**Comment by tfcollins** [comment]
Can you provide all the commands used and version of OS building on?
